File: java/server/src/main/java/com/github/fernthedev/lightchat/server/event/AuthenticateRequestEvent.java
package com.github.fernthedev.lightchat.server.event;

import com.github.fernthedev.lightchat.core.api.event.api.HandlerList;
import com.github.fernthedev.lightchat.server.security.AuthenticationManager;

/**
 * Called when client has successfully established a secure and valid connection
 */
public class AuthenticateRequestEvent extends AuthenticateEvent {


    private static final HandlerList handlers = new HandlerList();

    public AuthenticateRequestEvent(AuthenticationManager.PlayerInfo playerInfo) {
        super(playerInfo);
    }

    public AuthenticateRequestEvent(AuthenticationManager.PlayerInfo playerInfo, boolean async) {
        super(playerInfo, async);
    }


    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: java/server/src/main/java/com/github/fernthedev/lightchat/server/event/AuthenticateRequestEvent.kt
package com.github.fernthedev.lightchat.server.event

import com.github.fernthedev.lightchat.core.api.event.api.HandlerList
import com.github.fernthedev.lightchat.server.security.AuthenticationManager.PlayerInfo

/**
 * Called when client has successfully established a secure and valid connection
 */
class AuthenticateRequestEvent : AuthenticateEvent {
    constructor(playerInfo: PlayerInfo) : super(playerInfo)
    constructor(playerInfo: PlayerInfo, async: Boolean) : super(playerInfo, async)

    companion object {
        val handlerList = HandlerList()
    }
}